src/source.cpp

```diff
@@ -153,17 +153,14 @@ Bank SourceDistribution::sample() const
   static int n_accept = 0;
   while (!found) {
     // Set particle type
-    site.particle = static_cast<int>(particle_);
+    site.particle = particle_;
 
     // Sample spatial distribution
-    Position r = space_->sample();
-    site.xyz[0] = r.x;
-    site.xyz[1] = r.y;
-    site.xyz[2] = r.z;
+    site.r = space_->sample();
 
     // Now search to see if location exists in geometry
     int32_t cell_index, instance;
-    int err = openmc_find_cell(site.xyz, &cell_index, &instance);
+    int err = openmc_find_cell(&site.r.x, &cell_index, &instance);
```

Comment:
Not a fan of assuming that the struct has the same spacing as an array.  It probably does, but I doubt it's guaranteed by the standards.

Reply:
Couldn't slip that one pass you, huh? :) I thought about adding a `static_assert` to ensure there's no extraneous padding in the struct, but it seems safer to just convert this to an array before passing so I'll do that.

Reply:
After all those C++ PRs you'll need a lot more than 3,000 diffs to hide ad hoc coding :p